StRoot/StFttClusterMaker/StFttClusterMaker.cxx

```diff
@@ -0,0 +1,429 @@
+            nStripsHit++;
+        }
+        if ( kFttVertical   == so ){
+            vStripsPerRob[ rob ].push_back(hit);
+            nStripsHit++;
+        }
+        if ( kFttDiagonalH   == so ){
+            dhStripsPerRob[ rob ].push_back(hit);
+            nStripsHit++;
+        }
+        if ( kFttDiagonalV   == so ){
+            dvStripsPerRob[ rob ].push_back(hit);
+            nStripsHit++;
+        }
+    } // loop on hit
+
+    size_t nClusters = 0;
+    LOG_INFO << "nStripsHit = " << nStripsHit << endm;
+    if ( nStripsHit > 0 ){ // could make more strict?
+        for ( UChar_t iRob = 1; iRob < StFttDb::nRob+1; iRob++ ){
+
+            auto hClusters = FindClusters( hStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : hClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+            auto vClusters = FindClusters( vStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : vClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+            auto hdClusters = FindClusters( dhStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : hdClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+            auto vdClusters = FindClusters( dvStripsPerRob[iRob] );
+            // Add them to StEvent  
+            for ( StFttCluster * clu : vdClusters ){
+                mFttCollection->addCluster( clu );
+                nClusters++;
+            }
+        } // loop on iRob
+    } // nStripsHit
+    LOG_INFO << "Found " << nClusters << " clusters this event" << endm;
+
+    return kStOk;
+} // Make
+
+void StFttClusterMaker::InjectTestData(){
+    mFttCollection->rawHits().clear();
+
+    StFttRawHit *hit = new StFttRawHit( 1, 1, 1, 1, 1, 55, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 23, kFttHorizontal ); // LEFT 2
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 90, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 24, kFttHorizontal ); // LEFT 1
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 60, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 27, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 95, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 25, kFttHorizontal ); // CENTER
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 93, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 26, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+
+    hit = new StFttRawHit( 1, 1, 1, 1, 1, 19, 1, 1, 0 );
+    hit->setMapping( 1, 1, 1, 28, kFttHorizontal );
+    mFttCollection->addRawHit( hit );
+} // InjectTestData
+
+
+bool StFttClusterMaker::PassTimeCut( StFttRawHit * hit ){
+    return (abs( hit->time() ) <=3); 
```

Comment:
Useful to make this cut value a DB parameter as well?

Reply:
Addressed: 3686a6eea3b300c95180bdcee0ad698b5763a88c